python-packages/middlewares/setup.py

```diff
@@ -0,0 +1,199 @@
+        )
+
+        for lint_command in lint_commands:
+            print(
+                "Running lint command `", " ".join(lint_command).strip(), "`"
+            )
+            subprocess.check_call(lint_command)  # nosec
+
+
+class CleanCommandExtension(clean):
+    """Custom command to do custom cleanup."""
+
+    def run(self):
+        """Run the regular clean, followed by our custom commands."""
+        super().run()
+        rmtree("dist", ignore_errors=True)
+        rmtree(".mypy_cache", ignore_errors=True)
+        rmtree(".tox", ignore_errors=True)
+        rmtree(".pytest_cache", ignore_errors=True)
+        rmtree("src/0x_middlewares.egg-info", ignore_errors=True)
+
+
+class TestPublishCommand(distutils.command.build_py.build_py):
+    """Custom command to publish to test.pypi.org."""
+
+    description = (
+        "Publish dist/* to test.pypi.org." "Run sdist & bdist_wheel first."
+    )
+
+    def run(self):
+        """Run twine to upload to test.pypi.org."""
+        subprocess.check_call(  # nosec
+            (
+                "twine upload --repository-url https://test.pypi.org/legacy/"
+                + " --verbose dist/*"
+            ).split()
+        )
+
+
+class PublishCommand(distutils.command.build_py.build_py):
+    """Custom command to publish to pypi.org."""
+
+    description = "Publish dist/* to pypi.org. Run sdist & bdist_wheel first."
+
+    def run(self):
+        """Run twine to upload to pypi.org."""
+        subprocess.check_call("twine upload dist/*".split())  # nosec
+
+
+class PublishDocsCommand(distutils.command.build_py.build_py):
+    """Custom command to publish docs to S3."""
+
+    description = (
+        "Publish docs to "
+        + "http://0x-middlewares-py.s3-website-us-east-1.amazonaws.com/"
+    )
+
+    def run(self):
+        """Run npm package `discharge` to build & upload docs."""
+        subprocess.check_call("discharge deploy".split())  # nosec
+
+
+with open("README.md", "r") as file_handle:
+    README_MD = file_handle.read()
+
+
+setup(
+    name="0x-middlewares",
+    version="1.0.1",
+    description="Web3 middlewares for 0x applications",
+    long_description=README_MD,
+    long_description_content_type="text/markdown",
+    url="https://github.com/0xproject/0x-monorepo/python-packages/middlewares",
+    author="Michael Huang",
+    author_email="michaelhly@users.noreply.github.com",
+    cmdclass={
+        "clean": CleanCommandExtension,
+        "lint": LintCommand,
+        "test": TestCommandExtension,
+        "test_publish": TestPublishCommand,
+        "publish": PublishCommand,
+        "publish_docs": PublishDocsCommand,
+    },
+    install_requires=[
+        "eth_account",
+        "eth_keys",
+        "hypothesis>=3.31.2",  # HACK! this is web3's dependency!
+        # above works around https://github.com/ethereum/web3.py/issues/1179
+        "mypy_extensions",
+    ],
+    extras_require={
+        "dev": [
+            "0x-contract-addresses",
```

Comment:
There are a bunch of requirements in this list that should actually be in `install_requires`, not `extras_require['dev']`.  Anything that clients need at runtime should be in `install_requires`; that includes contract addresses, order utils, web3.  Also, double check that your tox.ini's list of dependencies in `[testenv:run_tests_against_test_deployment]` matches `install_requires`.  (Should be an exact match, plus pytest.)

Reply:
The classes you have mentioned are only used in the tests not the middleware itself. Do they need to be in `requires`?

Reply:
Talked to @michaelhly about this.  You're right, if they're only used in tests then `extras_require[dev]` is indeed the right place for them.